Description: Fix tools/rbinstall.rb to work with newer rubygems.
Author:      Alex Tomlins <alex.tomlins@digital.cabinet-office.gov.uk>
Last-Update: 2013-10-21
diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 5162498..1acaedf 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -553,11 +553,18 @@ end
 
 # :stopdoc:
 module Gem
+  if defined?(BasicSpecification)
+    remove_const(:BasicSpecification)
+  end
+
+  class BasicSpecification < OpenStruct
+  end
+
   if defined?(Specification)
     remove_const(:Specification)
   end
 
-  class Specification < OpenStruct
+  class Specification < BasicSpecification
     def initialize(*)
       super
       yield(self) if defined?(yield)
